src/state_root_anchor_service.js

```diff
@@ -0,0 +1,95 @@
+
+const assert = require('assert');
+const interval = require('interval-promise');
+const Mosaic = require('@openstfoundation/mosaic.js');
+
+/** @todo Move into utils. */
+function isObject(obj) {
+  return typeof obj === 'object' && obj !== null && !Array.isArray(obj);
+}
+
+class StateRootAnchorService {
+  constructor(
+    direction,
+    anchorBlockDelay,
+    mosaic,
+  ) {
+    if (direction === 'origin') {
+      this.source = 'origin';
+      this.target = 'auxiliary';
+    } else if (direction === 'auxiliary') {
+      this.source = 'auxiliary';
+      this.target = 'origin';
+    } else {
+      throw new Error(`Unexpected value (${direction}) for direction. `
+        + 'Expected values are: origin, auxiliary.');
+    }
+
+    assert(Number.isInteger(anchorBlockDelay));
+    assert(anchorBlockDelay > 0);
```

Comment:
why?

Reply:
In my mind it should be a positive value. However, if you think we should check >= 0, let me know.

Reply:
I think delaying zero blocks is technically possible. And if we say `> 0`, then why not `> 1`, etc...

Reply:
Agree. >= 0.